test/samples/meson.build

```diff
@@ -68,7 +68,43 @@ foreach name, cmd: samples
     env = name == 'locale.mkv' ? 'LANG=C' : []
     testname = 'libmpv-test-' + name
     exe = executable(testname, '../libmpv_test_track_selection.c',
-                     dependencies: libmpv_dep)
+                     c_args: libmpv_flags, dependencies: libmpv_dep)
     test(testname, exe, args: [name, target.full_path()],
          depends: target, env: env, suite: 'libmpv')
 endforeach
+
+
+sub_sdh = join_paths(source_root, 'test', 'samples', 'sub_sdh.srt')
+sub_mixed = join_paths(source_root, 'test', 'samples', 'sub_mixed.srt')
+
+sdh_samples = {
+    'sdh_default.mkv':
```

Comment:
Ha, you made it end-to-end tested. I was kinda thinking just calling the filter() function on some strings would be enough, but this works too.